pyqtgraph/palette.py

```diff
@@ -0,0 +1,414 @@
+            'w': QtCore.Qt.white , 's': QtCore.Qt.gray, 'k': QtCore.Qt.black  ,
+            'l': QtCore.Qt.lightGray, 'd': QtCore.Qt.darkGray
+        }
+        if not self.dark: # darken some colors for light mode
+            colors['c'] = QtCore.Qt.darkCyan
+            colors['g'] = QtCore.Qt.darkGreen
+            colors['y'] = QtCore.Qt.darkYellow
+            colors['m'] = QtCore.Qt.darkMagenta
+        self.add(colors)
+        colors = { 
+            'p'+str(idx) : name 
+            for idx, name in enumerate( ('gr_fg', 'y','r','m','b','c','g','s') ) 
+        }
+        self.add(colors)
+        
+    def setMonochrome(self, color='green'):
+        """ 
+        Updates graph colors with a set based on 'monochrome' color map,
+        imitating a monochrome computer screen
+        ==============  =================================================================
+        **Arguments:**
+        color           Primary color description. Can be one of predefined identifiers
+                        'green', 'amber', 'blue'
+                        or a tuple of relative R,G,B contributions in range 0.0 to 1.0
+        ==============  =================================================================
+    """
+        print('preparing monochrome map for',str(color))
+        cmap = colormap.makeMonochrome(color)
+        if cmap is None: 
+            raise ValueError("Failed to generate color for '"+str(color)+"'")
+        self.sampleColorMap( cmap=cmap, start=1.0, step=-1/8 ) # assign bright to dark, don't go all the way to background.
+        # define colors 'm0' (near-black) to 'm8' (near-white):
+        self.sampleColorMap( n_colors=10, cmap=cmap, step=1/9, prefix='m' ) 
+        colors =  {
+            'gr_bg' : 'm0', 'gr_fg' : 'm4',
+            'gr_txt': 'm6', 'gr_acc': 'm6',
+            'gr_hlt': 'm8', 'gr_reg': ('m1',192),
+            'k': 'm0', 'd': 'm1', 's': 'm3', 'l': 'm6', 'w': 'm9'
+        }        
+        self.add( colors )        
+        # make a dictionary of plot colors (except darkest and lightest) to emulate primary colors:
+        needed = { # int RGB colors that we are looking to emulate:
+            'b': np.array((  0,  0,255)), 'c': np.array((  0,160,160)), 
+            'g': np.array((  0,255,  0)), 'y': np.array((160,160,  0)),
+            'r': np.array((255,  0,  0)), 'm': np.array((160,  0,160))
+        }
+        
+        # project legacy colors onto monochrome reference to assigne them somewhat logically
+        ref_color = np.array( self.palette['m5'].getRgb()[:3] )
+        brightness = [
+            (np.sum( ref_color * needed[key] ), key) 
+            for key in needed
+        ]
+        brightness = sorted(brightness, key=lambda brightness: brightness[0])
+        # print( brightness )
+        avail = ('m3','m4','m5','m6','m7','m8')
+        colors = {}
+        for idx, (value, key) in enumerate(brightness):
+            colors[key] = avail[idx]
```

Comment:
dict comprehension time!

```python
colors = {key: avail[idx] for idx, (_, key) in enumerate(brightness}
```

Actually, relooking at this, probably better off... (was guessing on variable names, please make better choices than I did 😆  )

```python
colors = {key: avai for _, key, avai in zip(brightness, avail)
```

Reply:
I think I'll change this to
```python
        choice = ('m3','m4','m5','m6','m7','m8') # 6 candidates for 6 needed colors
        colors = {
            key: choice[idx] # assign in order of dark to bright
            for idx, (_, key) in enumerate(brightness)
        }
```
It's not quite as compact, but I personally find it easier to see the ordering when the index is explicitly present.